### DISCUSSION TITLE
Composer update error in mac

### DISCUSSION THREAD

**Discussion by lizadelacb** [composer/composer#10044] [Q&A / Support]
I tried to update composer in a mac and it shows me this error:

Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 4096 bytes) in phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Solver.php on line 201

Does anyone have any clue how can I fix it?

**Comment by Seldaek** [comment]
See https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for troubleshooting help.

Also check the output of `composer -V`, if you are still using composer 1.x make sure you update to composer2 using `composer self-update --2`, as it uses a lot less memory.
